core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks.go

```diff
@@ -0,0 +1,828 @@
+    am.interopChaincodeId = interopChaincodeId
+}
+
+
+// Ledger transaction (invocation) functions
+
+func (am *AssetManagement) AddFungibleAssetCount(stub shim.ChaincodeStubInterface, assetType string, numUnits int) (bool, error) {
+    var errorMsg string
+
+    if len(am.interopChaincodeId) == 0 {
+        errorMsg = "Interoperation chaincode ID not set. Run the 'Configure(...) function first."
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+
+    if len(assetType) == 0 {
+        errorMsg = "Empty asset type"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if numUnits <= 0 {
+        errorMsg = "Invalid number of asset units"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    iccResp := stub.InvokeChaincode(am.interopChaincodeId, [][]byte{[]byte("AddFungibleAssetCount"), []byte(assetType), []byte(strconv.Itoa(numUnits))}, "")
+    fmt.Printf("Response from Interop CC: %+v\n", iccResp)
+    if iccResp.GetStatus() != shim.OK {
+        return false, fmt.Errorf(string(iccResp.GetPayload()))
+    }
+    fmt.Printf("%d units of asset type %s added to the record\n", numUnits, assetType)
+    return true, nil
+}
+
+func (am *AssetManagement) LockAsset(stub shim.ChaincodeStubInterface, assetAgreement *common.AssetAgreement, lockInfo *common.AssetLock) (bool, error) {
+    var errorMsg string
+
+    if len(am.interopChaincodeId) == 0 {
+        errorMsg = "Interoperation chaincode ID not set. Run the 'Configure(...) function first."
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+
+    if (lockInfo.LockMechanism != common.LockMechanism_HTLC) {
+        log.Errorf("Unsupported lock mechanism: %+v", lockInfo.LockMechanism)
+        return false, fmt.Errorf("Unsupported lock mechanism: %+v", lockInfo.LockMechanism)
+    }
+    lockInfoHTLC := &common.AssetLockHTLC{}
+    if len(lockInfo.LockInfo) == 0 {
+        log.Error("Empty lock info")
+        return false, fmt.Errorf("Empty lock info")
+    }
+    err := proto.Unmarshal(lockInfo.LockInfo, lockInfoHTLC)
+    if err != nil {
+        log.Error(err.Error())
+        return false, err
+    }
+    if lockInfoHTLC.TimeSpec != common.AssetLockHTLC_EPOCH {
+        log.Error("Only EPOCH time is supported at present")
+        return false, fmt.Errorf("Only EPOCH time is supported at present")
+    }
+
+    if len(assetAgreement.Type) == 0 {
+        errorMsg = "Empty asset type"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if len(assetAgreement.Id) == 0 {
+        errorMsg = "Empty asset ID"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if len(assetAgreement.Recipient) == 0 {
+        errorMsg = "Empty lock recipient"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if len(lockInfoHTLC.Hash) == 0 {
```

Comment:
perhaps we can bring this next to other validation checks on `lockInfoHTLC` (i.e. move to line 98)

Reply:
Done. Here and in other places.